docs/basics/101-130-yodaproject.rst

```diff
@@ -0,0 +1,564 @@
+  stamp for each modification thanks to :command:`datalad save`.
+- The raw dataset knows where the data came from thanks to
+  :command:`datalad download-url`.
+- The subdataset is linked to the superdataset thanks to
+  :command:`datalad install -d`.
+- The :command:`datalad run` command took care of linking the outputs of your
+  analysis with the script and the input data it was generated from.
+
+Let's take a look at the history of the ``midterm_project`` analysis
+dataset:
+
+.. runrecord:: _examples/DL-101-130-114
+   :language: console
+   :workdir: dl-101/DataLad-101/midterm_project
+
+   $ git log
+
+"Wow, this is so clean an intuitive!" you congratulate yourself. "And I think
+this was and will be the fastest I have ever completed a midterm project!"
+But what is still missing is a human readable description of your dataset.
+The YODA procedure kindly placed a ``README.md`` file into the root of your
+dataset that you can use for this [#f4]_.
+
+.. runrecord:: _examples/DL-101-130-115
+   :language: console
+   :workdir: dl-101/DataLad-101/midterm_project
+
+   # with the >| redirection we are replacing existing contents in the file
+   $ cat << EOT >| README.md
+
+   # Midterm YODA Data Analysis Project
+
+   ## Dataset structure
+
+   - All inputs (i.e. building blocks from other sources) are located in input/.
+   - All custom code is located in code/.
+   - All results (i.e., generated files) are located in output/.
+
+   EOT
+
+.. runrecord:: _examples/DL-101-130-116
+   :language: console
+   :workdir: dl-101/DataLad-101/midterm_project
+
+   $ datalad status
+
+.. runrecord:: _examples/DL-101-130-117
+   :language: console
+   :workdir: dl-101/DataLad-101/midterm_project
+
+   $ datalad save -m "Provide project description" README.md
+
+To be extra helpful we will also create a README.md inside of ``outputs``
+that tells others about the nature of the result files:
+
+.. runrecord:: _examples/DL-101-130-118
+   :language: console
+   :workdir: dl-101/DataLad-101/midterm_project
+
+   $ cat << EOT > output/README.md
+     This directory contains the analysis outputs.
+
+     - ``output/prediction_report.csv`` contains the main classification
+       metrics.
+     - ``output/pairwise_relationships.png`` is a plot of the relations
+       between features.
+
+   EOT
+
+.. index:: datalad command; save --to-git
+
+One feature of the YODA procedure was that it configured certain files (for
+examples everything inside of ``code/`` and the ``README.md`` file in the
+root of the dataset) to be saved in Git instead of Git-annex. This was the
+reason why the ``README.md`` in the root of the dataset was easily modifyable.
+However, such a configuration does not exist for the ``README.md`` file we
+are creating in ``output``. This means, should we :command:`datalad save`
+this file, it will be annexed -- an inconvenience, given that this file is
+small enough to be handled by Git.
```

Comment:
Yeah, this is what I mean. It is better to consider the entire dataset as output, and then the one top-level README can contain everything there is to know. Just my 2ct